homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,465 @@
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO224HPM25,
+        sensor_type=MAIN_DATA,
+        name="CO2 24H PM2.5 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2PM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM10",
+        device_class=SensorDeviceClass.PM10,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2RTPM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM10 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO224HPM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 24H PM10 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNING,
+        sensor_type=MAIN_DATA,
+        name="Thunder Last Distamce",
+        icon="mdi:lightning-bolt",
+        native_unit_of_measurement=UnitOfLength.MILES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNINGTIME,
+        sensor_type=MAIN_DATA,
+        name="Thunder Last Timestamp",
+        icon="mdi:lightning-bolt",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNINGNUM,
+        sensor_type=MAIN_DATA,
+        name="Thunder Daily Count",
+        icon="mdi:lightning-bolt",
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Wittiot sensor entities based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[MainDevWittiotSensor] = []
+    # Main Device Data
+    for desc in SENSOR_DESCRIPTIONS:
+        if desc.key in coordinator.data:
+            sensors.append(
+                MainDevWittiotSensor(
+                    coordinator,
+                    entry.data[CONF_HOST],
+                    entry.data[DEVICE_NAME],
+                    desc,
+                )
+            )
+
+    async_add_entities(sensors, True)
+
+
+class MainDevWittiotSensor(
+    CoordinatorEntity[WittiotDataUpdateCoordinator], SensorEntity
+):
+    """Define an Local sensor."""
+
+    _attr_has_entity_name = True
+    entity_description: WittiotSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WittiotDataUpdateCoordinator,
+        host: str,
+        devname: str,
+        description: WittiotSensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{devname}_{MAIN_DATA}")},
```

Comment:
In this case I would just omit the MAIN_DATA
```suggestion
            identifiers={(DOMAIN, devname)},
```